### PR TITLE
feat: upgrade otel to v1.15.1

### PR DESCRIPTION
**Pull request by k4leung4**
- [x] Do only one thing
- [x] Non breaking API changes
- [x] Tested

### What did this pull request do?

Due to the breaking changes of OpenTelemetry, we need to upgrade this package to keep using it with newer versions of the SDK.

### CODE REVIEW

**Comment by jinzhu** [comment]
Hi @k4leung4 

Can you help to fix the CI? thank you.

**Comment by zstone12** [comment]
LGTM

File: go.mod [hunk 1]
```diff
@@ -1,52 +1,53 @@
...
-	go.opentelemetry.io/otel/sdk/metric v0.36.0
-	go.opentelemetry.io/otel/trace v1.13.0
-	gorm.io/driver/sqlite v1.3.2
-	gorm.io/gorm v1.23.5
+	go.opentelemetry.io/otel/metric v0.38.1
+	go.opentelemetry.io/otel/sdk v1.15.1
+	go.opentelemetry.io/otel/sdk/metric v0.38.1
+	go.opentelemetry.io/otel/trace v1.15.1
+	gorm.io/driver/sqlite v1.5.0
+	gorm.io/gorm v1.25.0
```
Comment (diligiant) [inline go.mod:23]: @zstone12 is bumping logrus & gorm versions ok with you? While I love it, I don't know if you prefer this in another PR as it doesn't seem absolutely necessary.

File: go.mod [hunk 2]
```diff
@@ -1,52 +1,53 @@
...
+	go.opentelemetry.io/contrib/instrumentation/runtime v0.41.0
+	go.opentelemetry.io/contrib/propagators/b3 v1.16.0
+	go.opentelemetry.io/contrib/propagators/jaeger v1.16.0
+	go.opentelemetry.io/contrib/propagators/opencensus v0.41.0
+	go.opentelemetry.io/contrib/propagators/ot v1.16.0
+	go.opentelemetry.io/otel v1.15.1
+	go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc v0.38.1
+	go.opentelemetry.io/otel/exporters/otlp/otlptrace v1.15.1
+	go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc v1.15.1
 	go.opentelemetry.io/otel/exporters/stdout/stdouttrace v1.13.0
```
[the remark this answers is not in the document]
Comment (k4leung4) [reply go.mod]: no reason, i guess i missed that one.
pushed again to update that as well

[the remark this answers is not in the document]
Comment (diligiant) [reply go.mod]: sorry for not being exhaustive at first: `go list` tells that otel's instrumentation/runtime & the propagators can be bumped: they were published the same day as otel/sdk so I'd bet it's a consistent release.

File: go.mod (hunk 1 above)
Comment (zstone12) [reply go.mod:23]: I don't have a better preference, bump or not is fine:D

**Pull request merged**